easybuild/scripts/ld_wrapper.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+#
+# Chandan Basu, cbasu@nsc.liu.se.
+# Consider this piece of script GPL.
+#
+# Time-stamp: <2015-02-12 16:00:00>
+
+# This is a linker-wrapper.
+#
+# The wrapped linker gets lib paths from compiler
+# (e.g., gcc, gfortran, icc, ifort etc.) and adds
+# rpaths to binary.
+
+SCRIPTNAME=$(basename $0)
+ALLSCRIPTS=( $( which -a $SCRIPTNAME ) )
+$EB_LD_VERBOSE && echo "found scriptname: $SCRIPTNAME allscripts: ${ALLSCRIPTS[1]}"
+LDORIG=${ALLSCRIPTS[1]}
+LDWRAPPER=$EB_LD_FLAG
+LINKER=$EB_LINKER_NAME   ## not yet implemented
+EB_LD_VERBOSE=${EB_LD_VERBOSE:-false}
+
+
+$EB_LD_VERBOSE && echo "INFO: linking with rpath "
+
+  if [ -z "$LDWRAPPER" ] || [ "$LDWRAPPER" == 0 ]; then
+## call the system linker
+    $LDORIG "$@"
+    exit
+  elif [ "$LDWRAPPER" == 1 ]; then
+## Default rpath-ing option.
+    exclude_lib_paths=("/lib" "/lib64" "/usr" "/home" "/tmp" "/opt" "/proj")
+  elif [ "$LDWRAPPER" == 2 ]; then
+## More aggressive rpath-ing
+    exclude_lib_paths=("/tmp" "/opt")
+  fi
+
+  L=""
+  lib_array=()
+  dir=`pwd`
+  for (( i=${#@}; i >= 0; i-- )); do
+    if [[ ${!i} == "--enable-new-dtags" ]]; then
+	## we are removing this flag if passed as it creates a copy of rpath to runpath.
```

Comment:
Aren't we at the mercy of what the default is set to here? We probably should (always) be adding `--disable-new-dtags` to the arguments as well as removing `--enable-new-dtags`